client/packages/system/src/Patient/VaccinationCards/ListView.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { FC } from 'react';
+import {
+  TableProvider,
+  DataTable,
+  useColumns,
+  createTableStore,
+  NothingHere,
+  useFormatDateTime,
+  ColumnAlign,
+  useTranslation,
+  ProgramEnrolmentSortFieldInput,
+  useUrlQueryParams,
+  useNavigate,
+  RouteBuilder,
+} from '@openmsupply-client/common';
+import {
+  ProgramEnrolmentRowFragment,
+  useProgramEnrolments,
+} from '@openmsupply-client/programs';
+import { usePatient } from '../api';
+import { createQueryParamsStore, useQueryParamsStore } from '@common/hooks';
+import { AppRoute } from '@openmsupply-client/config';
+
+const VaccinationCardListComponent: FC = () => {
+  const {
+    sort: { sortBy, onChangeSortBy },
+  } = useQueryParamsStore();
+
+  const {
+    queryParams: { page, first, offset },
+    updatePaginationQuery,
+  } = useUrlQueryParams();
+
+  const patientId = usePatient.utils.id();
+
+  const { data, isError, isLoading } = useProgramEnrolments.document.list({
```

Comment:
Maybe we should try and re-factor this API to the new pattern sometime soon?

Reply:
Agree! Figured if we make actual changes to the API, we can refactor those bits, and otherwise maybe in the refactor fortnight?